app.js

```diff
@@ -2,31 +2,74 @@ const express = require("express");
 const app = express();
 
 const endPoints = require("./endpoints.json");
-const { getTopics, getArticleById, getAllArticles } = require("./controller");
+const {
+  getTopics,
+  getArticleById,
+  getAllArticles,
+  getCommentsByArticleId,
+  postComments,
+  patchVotes,
+  deleteComment,
+} = require("./controller");
+
+app.use(express.json());
 
 app.get("/api/topics", getTopics);
 
 app.get("/api", (req, res) => {
   res.status(200).send(endPoints);
 });
 
-app.get("/api/articles", getAllArticles)
+app.get("/api/articles", getAllArticles);
 
 app.get("/api/articles/:article_id", getArticleById);
 
+app.get("/api/articles/:article_id/comments", getCommentsByArticleId);
+
+app.post("/api/articles/:article_id/comments", postComments);
+
+app.patch("/api/articles/:article_id", patchVotes);
+
+app.delete("/api/comments/:comment_id", deleteComment);
+
+app.use((err, req, res, next) => {
+  console.log(err.code)
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  next(err);
+});
 app.all("*", (req, res, next) => {
   res.status(404).send({ message: "not found" });
 });
 
 app.use((err, req, res, next) => {
```

Comment:
Might be nice to extract these error handlers into their own file at this stage to keep things tidy